stdlib/public/Darwin/Foundation/NSObject.swift

```diff
@@ -251,6 +251,36 @@ extension _KeyValueCodingAndObserving {
         }
     }
     
+    /// Overload for keyPath having a optional target value.
+    /// Solve https://bugs.swift.org/browse/SR-6066
+    public func observe<Value>(
+            _ keyPath: KeyPath<Self, Value?>,
+            options: NSKeyValueObservingOptions = [],
+            changeHandler: @escaping (Self, NSKeyValueObservedChange<Value?>) -> Void)
+        -> NSKeyValueObservation {
+        return NSKeyValueObservation(object: self as! NSObject, keyPath: keyPath, options: options) { (obj, change) in
+            let valueConverter = { (originValue: Any?) -> Value?? in
+                if let unwrappedValue = originValue {
+                    // NSNull is used by KVO to signal that the property value was nil.
+                    if unwrappedValue is NSNull {
+                        return .some(nil)
+                    } else {
+                        return .some(unwrappedValue as? Value)
```

Comment:
Unless `Value == NSNull.self`, the expression `.some(unwrappedValue as? Value)` will already evaluate to `.some(nil)` for the case where `unwrappedValue is NSNull`.

In the case where `Value == NSNull.self` there's no way to distinguish between the property actually containing `NSNull` or containing `nil` so either behavior works.

Given that, we can just remove this `if` and the special handling of `NSNull` entirely and just use `return .some(unwrappedValue as? Value)`. Though we should probably keep the comment to explain why we're even unwrapping this at all instead of just using `return unwrappedValue as? Value?`.